scipy/stats/_morestats.py

```diff
@@ -4013,14 +4041,81 @@ def directionalmean(samples, *, axis=0, normalize=True):
+    directional_mean = mean/np.linalg.norm(mean, axis=-1, keepdims=True)
     return directional_mean
+
+
+def directionalvar(samples, *, axis=0, normalize=True):
+    """
+    Computes the directional variance of a sample of vectors.
+
+    The directional variance serves is a measured of "directional spread" for
+    vector data. It should be used when the magnitude of the vector data
+    is irrelevant.
+
+    Parameters
+    ----------
+    samples : array_like
+        Input array. Must be at least two-dimensional, and the last axis of the
+        input must correspond with the dimensionality of the vector space.
+        When the input is exactly two dimensional, this means that each row
+        of the data is a vector observation.
+    axis : int, default: 0
+        Axis along which the directional mean is computed.
+    normalize: boolean, default: True
+        If True, normalize the input to ensure that each observation is a
+        unit vector. It the observations are already unit vectors, consider
+        setting this to False to avoid unnecessary computation.
+
+    Returns
+    -------
+    directionalvar : float
+        Directional variance.
+
+    See also
+    --------
+    circvar: circular variance; i. e. directional variance for 2D *angles*
+
+    Notes
+    -----
+    This uses the following definition of circular variance: ``1-R``, where
+    ``R`` is the mean resultant vector. The returned value is in the
+    range [0, 1], 0 standing for no variance, and 1 for a large variance.
+
+    This definition is appropriate for *directional* data (i.e. vector data
+    for which the magnitude of each observation is irrelevant) but not
+    for *axial* data (i.e. vector data for which the magnitude and *sign* of
+    each observation is irrelevant).
+
+    References
+    ----------
+    .. [1] Mardia, Jupp. (2000). *Directional Statistics*. Wiley.
+
+    Examples
+    --------
+    As an example, let us calculate the directional variance for 10 highly
+    concentrated vectors.
+
+    >>> import numpy as np
+    >>> from scipy.stats import directionalvar
+    >>> data = np.array([[0.911, -0.387, 0.14], # first observation
+    ...                  [0.972,0.233, -0.027], # second observation
+    ...                  [0.956, -0.104, -0.273],
+    ...                  [0.889, 0.218, -0.402],
+    ...                  [0.996, 0.076, 0.043],
+    ...                  [0.737, -0.104, -0.668],
+    ...                  [0.987, -0.09, 0.133],
+    ...                  [0.929, 0.127, -0.348],
+    ...                  [0.951, 0.2967, -0.09],
+    ...                  [0.919, -0.142, 0.367]])
+    >>> directionalvar(data)
+    0.06841124416403555
+
+    """
+    samples = _dirstats_preprocessing(samples, normalize, axis)
+    n_samples = samples.shape[0]
+    resultant_vector = samples.sum(axis=0)
+    r = np.linalg.norm(resultant_vector)
+    return 1 - r/n_samples
```

Comment:
I'm not sure how Mardia and Jupp describe the definition. But when you mention "mean resultant vector" in the definition, I would have expected this to be written as:
```suggestion
    mean_vector = samples.mean(axis=0)
    r = np.linalg.norm(mean_vector)
    return 1 - r
```
I think it should be written to match the reference, but on Wikipedia, at least:

![image](https://user-images.githubusercontent.com/6570539/183203125-4a088492-2e2f-42e0-96a0-e392118f2a49.png)

where:

![image](https://user-images.githubusercontent.com/6570539/183204696-0ecdcf19-50d6-42fc-831d-e8504e185519.png)

So it seems that in the definition of R, the number of samples is already taken into account.

Reply:
1 - `norm(sample_mean)` is not guaranteed to lie between 0 and 1 as the sample mean could be anything. Therefore I think that my implementation is correct.

Reply:
I think it's equivalent. If they are all in the same direction, say $\hat{v}$, the arithmetic mean of them will be $\hat{v}$. The magnitude $R=1$, and the variance is then 0.